tests/integration_tests/functional/test_vsock.py

```diff
@@ -215,3 +218,93 @@ def test_vsock_transport_reset(
+    uvm_nano, microvm_factory, bin_vsock_path, test_fc_session_root_path
+):
+    """
+    Vsock transport reset test.
+    """
+    test_vm = uvm_nano
+    test_vm.add_net_iface()
+    test_vm.api.vsock.put(vsock_id="vsock0", guest_cid=3, uds_path=f"/{VSOCK_UDS_PATH}")
+    test_vm.start()
+    test_vm.wait_for_up()
+
+    # Generate the random data blob file.
+    blob_path, blob_hash = make_blob(test_fc_session_root_path)
+    vm_blob_path = "/tmp/vsock/test.blob"
+
+    # Set up a tmpfs drive on the guest, so we can copy the blob there.
+    # Guest-initiated connections (echo workers) will use this blob.
+    _copy_vsock_data_to_guest(test_vm.ssh, blob_path, vm_blob_path, bin_vsock_path)
+
+    host_socket_path = os.path.join(
+        test_vm.path, f"{VSOCK_UDS_PATH}_{ECHO_SERVER_PORT}"
+    )
+    host_socat_commmand = [
+        "socat",
+        "-dddd",
+        f"UNIX-LISTEN:{host_socket_path},fork",
+        "STDOUT",
+    ]
+    host_socat = subprocess.Popen(
+        host_socat_commmand, stdout=subprocess.PIPE, stderr=subprocess.PIPE
+    )
+
+    # Give some time for host socat to create socket
+    time.sleep(0.5)
+    assert Path(host_socket_path).exists()
+    test_vm.create_jailed_resource(host_socket_path)
+
+    # Create a socat process in the guest which will connect to the host socat
+    guest_socat_commmand = f"tmux new -d 'socat - vsock-connect:2:{ECHO_SERVER_PORT}'"
+    test_vm.ssh.run(guest_socat_commmand)
+
+    # socat should be running in the guest now
+    code, _, _ = test_vm.ssh.run("pidof socat")
+    assert code == 0
+
+    # Create snapshot.
+    snapshot = test_vm.snapshot_full()
+    test_vm.resume()
+
+    # After `create_snapshot` + 'restore' calls, connection should be dropped
+    code, _, _ = test_vm.ssh.run("pidof socat")
+    assert code == 1
+
+    # Kill host socat as it is not useful anymore
+    host_socat.kill()
+    host_socat.communicate()
+
+    # Terminate VM.
+    metrics = test_vm.flush_metrics()
+    validate_fc_metrics(metrics)
+    test_vm.kill()
+
+    # Load snapshot.
+    vm2 = microvm_factory.build()
+    vm2.spawn()
+    vm2.restore_from_snapshot(snapshot, resume=True)
+    vm2.wait_for_up()
+
+    # After snap restore all vsock connections should be
+    # dropped. This means guest socat should exit same way
+    # as it did after snapshot was taken.
+    code, _, _ = vm2.ssh.run("pidof socat")
+    assert code == 1
+
+    # Start guest echo server.
+    path = start_guest_echo_server(vm2)
+
+    # Check that vsock device still works.
+    # Test guest-initiated connections.
+    path = os.path.join(vm2.path, make_host_port_path(VSOCK_UDS_PATH, ECHO_SERVER_PORT))
+    check_guest_connections(vm2, path, vm_blob_path, blob_hash)
+
+    # Test host-initiated connections.
+    path = os.path.join(vm2.jailer.chroot_path(), VSOCK_UDS_PATH)
+    check_host_connections(path, blob_path, blob_hash)
+    metrics = vm2.flush_metrics()
+    validate_fc_metrics(metrics)
```

Comment:
this is already covered by a different test

Reply:
Same here